### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by cilfonegabriel**
# Hi reviewer!
### General requirements
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    subject.comments = [
+      Comment.new(post: subject, author: subject.author, text: 'Hi Tom!'),
+      Comment.new(post: subject, author: subject.author, text: 'Hi Tom!'),
+      Comment.new(post: subject, author: subject.author, text: 'Hi Tom!'),
+      Comment.new(post: subject, author: subject.author, text: 'Hi Tom!'),
+      Comment.new(post: subject, author: subject.author, text: 'Hi Tom!'),
+      Comment.new(post: subject, author: subject.author, text: 'Hi Tom!')
+    ]
+    expect(subject.recent_comments.length).to eql(5)
+  end
```
Comment (omar-muhamad) [inline spec/models/post_spec.rb]: - [ ] Great job adding tests to user and post models, as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=equal%20to%20zero.-,Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.,-Need%20a%20big), Kindly add a test case for `update_post_counter` method too.

- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=equal%20to%20zero.-,Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.,-Need%20a%20big), Please also add tests to `comment` and `like` models too.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @cilfonegabriel 

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @cilfonegabriel , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has one of the linter checks failing, Kindly ensure all linter checks are passing.



Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Ismailco — APPROVED** [review]
### Hi @cilfonegabriel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ Clean code
✔️ Implement tests for all models
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**